### DISCUSSION TITLE
Adonis 5: Table Prefix

### DISCUSSION THREAD

**Discussion by mastermunj** [adonisjs/core#1358] [Help] [answered]
I am trying to have table prefix as 'tbl_', however, it gives me following TypeScript error.

My Config:

```ts
pg: {
  client: 'pg',
  connection: {
    host: Env.get('DB_HOST', '127.0.0.1') as string,
    port: Number(Env.get('DB_PORT', 5432)),
    user: Env.get('DB_USER', 'lucid') as string,
    password: Env.get('DB_PASSWORD', 'lucid') as string,
    database: Env.get('DB_NAME', 'lucid') as string,
  },
  healthCheck: true,
  prefix: 'tbl_',
},
```

The Error:

```
Type '{ client: "pg"; connection: { host: string; port: number; user: string; password: string; database: string; }; healthCheck: true; prefix: string; }' is not assignable to type 'SqliteConfig | MysqlConfig | PostgreConfig | OracleConfig | MssqlConfig'.
  Object literal may only specify known properties, and 'prefix' does not exist in type 'PostgreConfig'.
```

Am I doing something wrong?

**Comment by thetutlage** [comment]
Where in the docs you read that you can prefix tables?

**Comment by mastermunj** [comment]
I read it in v4.1 docs at https://adonisjs.com/docs/4.1/database#_table_prefixing and thought it will be available for v5 as well.

**Reply by thetutlage** [reply]
Ohhh okay. Nope, its not part of v5 yet. To be honest, I didn't even added it to the roadmap thinking the feature is not popular enough to be supported.

**Reply by mastermunj** [reply]
So the alternative is to set will be?

```ts
public static table = 'tbl_brands';
```

**Reply by thetutlage** [reply]
If it's about just models, then you can also use the ORM config to prefix the table name. Open `config/database.ts` file

```ts
  orm: {
    getTableName(model) {
      return `prefix_${plural(snakeCase(model.name))}`
    },
  },
```

And then import the following package to pluralize and snake case the model name. (have to install them as well)

```ts
import { plural } from 'pluralize'
import { snakeCase } from 'snake-case'
```

**Reply by mastermunj** [reply]
Will this reflect into `Model.table` when used anywhere?

**Reply by thetutlage** [reply]
Yes it will

**Reply by mastermunj** [reply]
It works! Thank you.

**Comment by mastermunj** [comment, marked as answer]
For others, the answer is https://github.com/adonisjs/core/discussions/1358#discussioncomment-43737
